Author: Colin Watson <cjwatson@ubuntu.com>
Description: Fix detection of floating point endianness.
 See https://lists.gnu.org/archive/html/bug-gnulib/2013-12/msg00104.html
 Based on a suggestion by Alan Modra.

--- a/gnulib-tests/test-isnanl.h
+++ b/gnulib-tests/test-isnanl.h
@@ -25,6 +25,12 @@
 #include "snan.h"
 #include "macros.h"
 
+#ifdef __FLOAT_WORD_ORDER__
+# define FLOAT_BIG_ENDIAN (__FLOAT_WORD_ORDER__ != __ORDER_LITTLE_ENDIAN__)
+#else
+# define FLOAT_BIG_ENDIAN (LDBL_EXPBIT0_WORD < NWORDS / 2)
+#endif
+
 int
 main ()
 {
--- a/gnulib-tests/test-signbit.c
+++ b/gnulib-tests/test-signbit.c
@@ -38,6 +38,12 @@ float zerof = 0.0f;
 double zerod = 0.0;
 long double zerol = 0.0L;
 
+#ifdef __FLOAT_WORD_ORDER__
+# define FLOAT_BIG_ENDIAN (__FLOAT_WORD_ORDER__ != __ORDER_LITTLE_ENDIAN__)
+#else
+# define FLOAT_BIG_ENDIAN (LDBL_EXPBIT0_WORD < NWORDS / 2)
+#endif
+
 static void
 test_signbitf ()
 {
